classes/local/store/object_file_system.php

```diff
@@ -421,6 +421,13 @@ public function xsendfile_file(stored_file $file): bool {
 
             return $this->redirect_to_presigned_url($contenthash, headers_list());
         }
+
+        if ($this->externalclient->support_presigned_urls()) {
+            $ranges = $this->get_valid_http_ranges($file->get_filesize());
+            if ($file->get_filesize() > 2097152 && $ranges) {
```

Comment:
can you make a const for this magic number and add a comment explaining it

Reply:
probably point to the php max temp / aws sdk comments

Reply:
Made a constant and added a comment with a link.
Didn't find how to get this php max temp on the fly. 
Looks like it's 2MB by default, but can be overriden via code. And there is no way to get current `maxmemory`.